test/jdk/com/sun/crypto/provider/KDF/HKDFKnownAnswerTests.java

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8331008
+ * @run main HKDFKnownAnswerTests
+ * @summary Tests for HKDF Expand and Extract Key Derivation Functions
+ * @enablePreview
+ */
+
+import javax.crypto.KDF;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.HKDFParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.Key;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.AlgorithmParameterSpec;
+import java.security.spec.InvalidParameterSpecException;
+import java.util.Arrays;
+import java.util.HexFormat;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+public class HKDFKnownAnswerTests {
+    public static class TestData {
+        public TestData(String name, String algStr, String ikmStr,
+                        String saltStr, String infoStr, int oLen,
+                        String expPrkStr,
+                        String expOkmStr) {
+            testName = Objects.requireNonNull(name);
+            algName = Objects.requireNonNull(algStr);
+            ikm = HexFormat.of().parseHex(Objects.requireNonNull(ikmStr));
+            if ((outLen = oLen) <= 0) {
+                throw new IllegalArgumentException(
+                    "Output length must be greater than 0");
+            }
+            expectedPRK = HexFormat.of().parseHex(Objects.requireNonNull(expPrkStr));
+            expectedOKM = HexFormat.of().parseHex(Objects.requireNonNull(expOkmStr));
+
+            // Non-mandatory fields - may be null
+            salt = (saltStr != null) ? HexFormat.of().parseHex(saltStr) : null;
+            info = (infoStr != null) ? HexFormat.of().parseHex(infoStr) : null;
```

Comment:
Instead of `null`, you should use new byte[0] for the default value when the corresponding string value is `null`. Because the `addSalt()` is always invoked regardless of the salt value.